### DISCUSSION TITLE
Improvements for Symbols window

### DISCUSSION THREAD

**Discussion by ArkadiuszMichalski** [Ekopalypse/NppLspClient#23] [Ideas]
It would be nice if Symbols windows had some basic features:
- search field (fuzzy variant recommended)
- ability to collapse/expand nested items
- ability to group global things by meaning (class/function/var etc.). It's easier to navigate over such things when they are not mixed.

You can check how it looks in [sourcecookifier](https://sourcecookifier.sourceforge.net/), because it was done nicely there.

I saw a question in some topic that you wanted to remove this window. I am against it, the built-in Function List is just a prosthesis and is not as good as some dedicated solutions. Nothing urgent, but some sensible panel with symbols for a given file is very useful.

**Comment by Ekopalypse** [comment]
> I saw a question in some topic that you wanted to remove this window.

More or less, yes, I thought about when I need to see the symbols in general and realized that I personally only use them when I don't know the exact name and therefore try to look it up. But dialog such as the completion dialog, which of course would have to be started manually, would be much more efficient here.
